canals/component/component.py

```diff
@@ -206,39 +207,55 @@ def run(self, data: <Input if defined, otherwise untyped>) -> <Output if defined
     return class_
 
 
-def _check_input(class_):
-    """
-    Check that the component's input respects all constraints
-    """
-    if not hasattr(class_, "Input") and not hasattr(class_, "input_type"):
-        raise ComponentError(
-            "Components must either have an Input dataclass or a 'input_type' property that returns such dataclass"
-        )
-    if hasattr(class_, "Input"):
-        if not is_dataclass(class_.Input):
-            raise ComponentError(f"{class_.__name__}.Input must be a dataclass")
-        if not hasattr(class_.Input, "_component_input"):
-            raise ComponentError(f"{class_.__name__}.Input must inherit from ComponentInput")
-        if (
-            hasattr(class_.Input, "_variadic_component_input")
-            and len(inspect.signature(class_.Input.__init__).parameters) != 2
-        ):
-            raise ComponentError("Variadic inputs can contain only one variadic positional parameter.")
-
-
-def _check_output(class_):
+# We do this to have some namespacing and also to make it clear that the methods decorated with
+# @component.input and @component.output must have their class decorated as @component.
+setattr(component, "input", _input)
+setattr(component, "output", _output)
```

Comment:
This is kinda ugly, I plan to make `@component` a class to make it more versatile.